server/src/test/java/keywhiz/service/providers/ClientAuthFactoryTest.java

```diff
@@ -193,37 +206,71 @@ public void clientWhenPrincipalAbsentThrows() {
     assertThat(factory.provide(request, httpServletRequest)).isEqualTo(client);
   }
 
-  @Test(expected = NotAuthorizedException.class)
-  public void rejectsXfcc_requesterAuthMissing() {
+  @Test public void returnsClientWhenClientPresent_fromXfccHeader_customSpiffeIdHeader() {
```

Comment:
Do we ever expect both the XFCC header and the custom SPIFFE ID header to be set? If so, Keywhiz should check that the same client is extracted from both headers (and deny access if they don't match). If not, Keywhiz should check that at most one header is set and deny access if both are set.

Reply:
Please make sure to have some tests with only the custom SPIFFE ID header set, so that we can be sure Keywhiz isn't falling back to the XFCC header path somehow (for instance, if the SPIFFE ID name mismatched) and still retrieving the client that way.

Reply:
I've updated the commit message to reflect the fact that the custom header should come with the XFCC header, and they may or may not match depending whether a client is accessing Keywhiz through a proxy.